frontend/packages/dev-console/src/components/topology2/Topology.tsx

```diff
@@ -0,0 +1,160 @@
+import { topologyModelFromDataModel } from './topology-utils';
+import layoutFactory from './layoutFactory';
+import widgetFactory from './widgetFactory';
+
+export interface TopologyProps {
+  data: TopologyDataModel;
+}
+
+const graphModel: Model = {
+  graph: {
+    id: 'g1',
+    type: 'graph',
+    layout: 'Force',
+  },
+};
+
+const Topology: React.FC<TopologyProps> = ({ data }) => {
+  const visRef = React.useRef<Visualization | null>(null);
+  const [model, setModel] = React.useState<Model>();
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
+  const [selectedEntity, setSelectedEntity] = React.useState<ElementEntity | undefined>();
+  const { topology } = data;
+
+  const onSelection = (ids: string[]) => {
+    setSelectedIds(ids);
+  };
+
+  if (!visRef.current) {
+    visRef.current = new Visualization();
+
+    visRef.current.registerLayoutFactory(layoutFactory);
+    visRef.current.registerWidgetFactory(widgetFactory);
+    visRef.current.addEventListener<SelectionEventListener>(SELECTION_EVENT, onSelection);
+  }
+
+  if (!model) {
+    const newModel = topologyModelFromDataModel(data);
+    visRef.current.fromModel(graphModel);
+    visRef.current.fromModel(newModel);
+    setModel(newModel);
+  }
+
+  React.useEffect(() => {
+    const newModel = topologyModelFromDataModel(data);
+    visRef.current.fromModel(newModel);
+    setModel(newModel);
+  }, [data]);
+
+  React.useEffect(() => {
+    let updatedSelection;
+    if (selectedIds[0]) {
+      try {
+        updatedSelection = visRef.current.getController().getEntityById(selectedIds[0]);
+      } catch (e) {
+        updatedSelection = undefined;
+      }
+    }
+
+    setSelectedEntity(updatedSelection);
+  }, [selectedIds, topology]);
+
+  React.useEffect(() => {
+    if (selectedEntity && isNodeEntity(selectedEntity)) {
+      setTimeout(() => {
+        visRef.current.getRoot().makeEntityVisible(selectedEntity);
+      }, 500);
+    }
+  }, [selectedEntity]);
+
+  const onSidebarClose = () => {
+    visRef.current.getController().fireEvent(SELECTION_EVENT, []);
+  };
+
+  const renderControlBar = () => {
+    return (
+      <TopologyControlBar
+        controlButtons={createTopologyControlButtons({
+          ...defaultControlButtonsOptions,
+          zoomInCallback: () => {
+            visRef.current.getRoot().scaleBy(4 / 3);
+          },
+          zoomOutCallback: () => {
+            visRef.current.getRoot().scaleBy(0.75);
+          },
+          fitToScreenCallback: () => {
+            visRef.current.getRoot().fit(80);
+          },
+          resetViewCallback: () => {
+            visRef.current.getRoot().reset();
+            visRef.current.getRoot().layout();
+          },
+          legend: false,
+        })}
+      />
+    );
+  };
+
+  const selectedItemDetails = () => {
+    if (isNodeEntity(selectedEntity)) {
+      if ((selectedEntity as BaseNodeEntity).isGroup()) {
```

Comment:
Do not cast to `BaseNodeEntity`:
```suggestion
      if (selectedEntity.isGroup()) {
```